### PR TITLE
Bug 1686736 - (RLB) Record the number of tasks queued in the pre-initialization buffer.

### PR DESCRIPTION
**Pull request by badboy**
Based on #1437.
The last 3 commits are the new implementation.

I went for an integration test to avoid having to deal with resetting the dispatcher and Glean.

### CODE REVIEW

**Comment by badboy** [comment]
I guess that will need a quick data review because it extends it to new clients?

**Comment by badboy** [comment]
From the original data request:

> 7) What populations will you measure?
> All products using Glean.


So I gues not.

File: glean-core/rlb/src/dispatcher/mod.rs [hunk 1]
```diff
@@ -107,6 +107,9 @@ struct DispatchGuard {
     /// Whether to queue on the preinit buffer or on the unbounded queue
     queue_preinit: Arc<AtomicBool>,
 
+    /// The number of items that were added to the queue after it filled up.
+    overflow_count: Arc<AtomicUsize>,
```
Comment (Dexterp37) [inline glean-core/rlb/src/dispatcher/mod.rs:111]: Can you kindly add a changelog entry?

**Review by Dexterp37 — APPROVED** [review]
r+wc

File: glean-core/rlb/src/dispatcher/mod.rs (hunk 1 above)
Comment (badboy) [reply glean-core/rlb/src/dispatcher/mod.rs:111]: I did ... then I rebased ... and lost it -_-

Comment (Dexterp37) [reply glean-core/rlb/src/dispatcher/mod.rs:111]: You almost had a perfect PR. Almost.

**Pull request merged**